core/domain/collection_services.py

```diff
@@ -134,23 +135,35 @@ def get_collection_from_model(collection_model, run_conversion=True):
     """
 
     # Ensure the original collection model does not get altered.
-    versioned_collection = {
+    versioned_collection_contents = {
         'schema_version': collection_model.schema_version,
-        'nodes': copy.deepcopy(collection_model.nodes)
+        'collection_contents':
+            copy.deepcopy(collection_model.collection_contents)
     }
 
+    # If collection_contents is empty, attempt to retrieve nodes data from nodes
+    # instead. This is temporary, and intended to not break backwards
+    # compatibility before the migration job is run.
+    # TODO(wxy): Remove this after collection migration is completed.
+    if versioned_collection_contents['collection_contents'] == {}:
```

Comment:
More pythonic:

``if not versioned_collection_contents['collection_contents']:``